server/plugins/BELAUGens.cpp

```diff
@@ -404,6 +404,19 @@ void AnalogIn_Ctor(AnalogIn* unit) {
 
 //////////////////////////////////////////////////////////////////////////////////////////////////
 
+bool AnalogOut_isValidPin(AnalogOut* unit, int newAnalogPin) {
+    BelaContext* context = unit->mWorld->mBelaContext;
+    bool isValid = (newAnalogPin > 0) && (newAnalogPin < context->analogOutChannels);
```

Comment:
make `newAnalogPin` an `unsigned  int` to avoid the `> 0` check. Also, `BelaContext::analogOutChannels` is itself an `unsigned int`

Reply:
Great, but how does this apply to when we read it from a float? Shall I just static_cast it to unsigned int?
Sorry I didn't really know what happens when you cast a negative float to unsigned int, but it looks like it goes to 0, so that would be indeed perfect

Reply:
>Sorry I didn't really know what happens when you cast a negative float to unsigned int,

Negative float? Same thing as a negative `int`: casting it to `unsigned int` (or passing it as an argument to something that takes an `unsigned int`) will just give a very large number.

No need to static_cast I think, unless the compiler complains ...

Reply:
So, it looks like it is implementation dependent. On some architectures it would give a large number, on others 0. It looks like it is 0 on arm, but anyway I would avoid casting to unsigned int, keep casting to signed int and check for > 0. What do you think?

Reply:
Not sure I understand why you'd need the negative float case ... currently you are parsing the argument as an `int`, and then passing it as an argument to a function with an `(... , unsigned int)` signature, so you are only casting from `int` to `unsigned int`, which should be well standardised ...